src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -1618,6 +1618,31 @@ HMICapabilitiesImpl::GetDefaultInitializedCapabilities() const {
   return default_initialized_capabilities_;
 }
 
+void HMICapabilitiesImpl::RemoveFromDefaultInitializedCapabilities(
+    hmi_apis::FunctionID::eType requested_interface) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (app_mngr_.IsHMICooperating()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Remove from default initialized capabilities skipped, "
+                  "because hmi_cooperating equal true already");
+    return;
+  }
+
+  auto it = find(default_initialized_capabilities_.begin(),
+                 default_initialized_capabilities_.end(),
+                 requested_interface);
+  if (it != default_initialized_capabilities_.end()) {
+    default_initialized_capabilities_.erase(it);
+    LOG4CXX_DEBUG(logger_,
+                  "Wait for " << default_initialized_capabilities_.size()
+                              << " responses");
+    if (default_initialized_capabilities_.empty()) {
+      app_mngr_.SetHMICooperating(true);
```

Comment:
Confusing that you setting HMICooperating(true) on calling RemoveFromDefaultInitializedCapabilities. Please extract if from here.

Reply:
@LuxoftAKutsan We need this line here, when HMI has responded to all the HMI capabilities requests, SDL Core should set IsHMICooperating to true.

Reply:
Right, we need it here, but naming of the function does not show that SDL will change internal state of app_mngr_.
Please call app_mngr_.SetHMICooperating(true); after calling the function

Reply:
@LuxoftAKutsan [add new function for call this one](https://github.com/smartdevicelink/sdl_core/pull/3269/files#diff-f67d217da60dca297da36f9d5dcbeac3R1641)